### PR TITLE
Bug 10675/create a rudder extension and third party integration chapter

### PR DESCRIPTION
**Pull request by fanf**
Issue https://www.rudder-project.org/redmine/issues/10675

### CODE REVIEW

File: 80_extension_and_integration/00_intro.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+== Rudder extension and integration with third party software
```
Comment (amousset) [inline 80_extension_and_integration/00_intro.txt]: required empty line here

File: 80_extension_and_integration/40_Rudder_Integration.txt [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+With that plugin, you can execute commands on node registered in Rudder, taking
+advantage of the groups defined for you policies.
+
+==== Ansible
+
+There is an https://github.com/ansible/ansible/blob/devel/contrib/inventory/rudder.py[inventory plugin]
+for Ansible that makes possible to use Rudder inventory (including groups, nodes,
+group ids, node ids, and node properties) as inventory for Ansible, for example
+for orchestration tasks on your platform.
```
Comment (amousset) [inline 80_extension_and_integration/40_Rudder_Integration.txt]: It is part of Ansible as of version 2.0.

**Comment by amousset** [comment]
Missing empty lines everywhere (I will fix the build process when I have time to avoid having to do this).

**Comment by fanf** [comment]
PR rebased

File: 80_extension_and_integration/00_intro.txt [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+== Rudder extension and integration with third party software
+
+Rudder was thought from the begining to be a good citizen in you infrastructure.
+Part of that good will intent is translated into the fact that everything is done
+so that Rudder is able to inter-operate well with the other parts of your
+infrastructure.
+
+For that, there is two mains way of integrating Rudder with your
+existing infrastructure components: you can either extends Rudder with plugins,
```
Comment (amousset) [inline 80_extension_and_integration/00_intro.txt]: extend

File: 80_extension_and_integration/00_intro.txt [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+or you can extends your existing tools or process to take advantage of Rudder.
```
Comment (amousset) [inline 80_extension_and_integration/00_intro.txt]: extend

File: 80_extension_and_integration/20_Rudder_plugins.txt [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+This plugin allows to add new tabs in Rudder "node details" page and display
+node specific files, stored in node-dedicated places. It also use a
+self-managed and hot-reloading configuration of its properties.
+
+===== Providing new authentication methods
+
+Rudder plugins can be used to provide new authentication methods. There is no
+open source version of such module, but at least a Radius Authentication Plugin
+exists
```
Comment (amousset) [inline 80_extension_and_integration/20_Rudder_plugins.txt]: exists.

File: 80_extension_and_integration/20_Rudder_plugins.txt [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+===== Providing a full new feature: rudder-plugin-datasources
+
+Link: https://github.com/Normation/rudder-plugin-datasources
+
+As we said, Rudder plugins can be quite powerfull. For example, the "Data sources
```
Comment (amousset) [inline 80_extension_and_integration/20_Rudder_plugins.txt]: powerful

File: 80_extension_and_integration/20_Rudder_plugins.txt [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+Plugin" bring a completly new feature to Rudder by allowing to configure external
```
Comment (amousset) [inline 80_extension_and_integration/20_Rudder_plugins.txt]: brings

Comment (amousset) [inline 80_extension_and_integration/20_Rudder_plugins.txt]: completely

File: 80_extension_and_integration/20_Rudder_plugins.txt [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+API data sources from which Rudder will get node properties. The plugin set-up
+its own data base table, come with his own UI (available in Rudder "administration"
```
Comment (amousset) [inline 80_extension_and_integration/20_Rudder_plugins.txt]: comes

Comment (amousset) [inline 80_extension_and_integration/20_Rudder_plugins.txt]: its own

File: 80_extension_and_integration/20_Rudder_plugins.txt [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+page), and interact with node properties.
```
Comment (amousset) [inline 80_extension_and_integration/20_Rudder_plugins.txt]: interacts

**Comment by fanf** [comment]
PR rebased

File: 80_extension_and_integration/40_Rudder_Integration.txt [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+It is part of Ansible as of version 2.0.
+
+===== iTop
+
+https://www.combodo.com/itop-193[iTop] is an Open Source CMDB solution. It allows
+to describe you IT services and analyse impact of problems.
+
+There is a prototype integration of iTop and Rudder which allow iTop to Rudder
```
Comment (amousset) [inline 80_extension_and_integration/40_Rudder_Integration.txt]: allows

**Comment by fanf** [comment]
PR rebased

File: 80_extension_and_integration/40_Rudder_Integration.txt [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+There is a prototype integration of iTop and Rudder which allows iTop to Rudder
+as a source of information about the server content (inventory) and current
+compliance level. With that integration, you can see in real time in your CMDB when
+a server managed by Rudder is drifting away from its expecting configuration, and
+use iTop to understand the impact of such a drift for your IT services.
+
+==== Integrate Rudder thanks to its APIs
+
+All the above plugins are using http://www.rudder-project.org/rudder-api-doc/[Rudder APIs]
+under the hood to operate or get data from Rudders. Rudder APIs are as powerfull
```
Comment (amousset) [inline 80_extension_and_integration/40_Rudder_Integration.txt]: powerful

File: 80_extension_and_integration/40_Rudder_Integration.txt [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+- manage (create, update, delete) groups, directives, rules and parameters,
+- interact with the Techniques library,
+- get compliance details about a node or a rule, with a parameterized depth of
+  information,
+- manage change requests.
+
+And of course, any plugin can provide new API endpoints, as is doing the
+data source plugin.
+
+These API can also to automate Rudder action, like node acceptation or compliance
```
Comment (amousset) [inline 80_extension_and_integration/40_Rudder_Integration.txt]: can also be used to

File: 80_extension_and_integration/40_Rudder_Integration.txt [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+exporte and archiving for nodes.
```
Comment (amousset) [inline 80_extension_and_integration/40_Rudder_Integration.txt]: export

**Comment by fanf** [comment]
PR rebased

**Comment by amousset** [comment]
This should target 4.1

**Comment by fanf** [comment]
PR replaced by https://github.com/Normation/rudder-doc/pull/327

**Pull request closed without merging**